src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerConfigOptions.cs

```diff
@@ -23,6 +22,11 @@ public abstract class AnalyzerConfigOptions
         /// Get an analyzer config value for the given key, using the <see cref="KeyComparer"/>.
         /// </summary>
         public abstract bool TryGetValue(string key, out string value);
+
+        /// <summary>
+        /// Get the keys of the defined options.
+        /// </summary>
+        public abstract ImmutableArray<string> Keys { get; }
```

Comment:
@jasonmalinowski @agocke Does this seem like a workable solution to enumerating the options? Now that the OptionSet is part of the solution snapshot. We can work on combining the solution options with the analyzer options in a follow up PR.

Reply:
After talking offline to Andy, I updated to `IEnumerable<string> Keys`.